src/PrimPlane.h

```diff
@@ -29,16 +29,18 @@ class CPrimPlane : public IPrim
 	virtual bool intersect(Ray& ray) const override
 	{
 		float dist = (m_origin - ray.org).dot(m_normal) / ray.dir.dot(m_normal);
-		if (dist < Epsilon || isinf(dist) || dist > ray.t) return false;
+		if (dist < Epsilon || isinf(dist) || dist > ray.t)return false;
 		
 		ray.t = dist;
+        //ray.hit = this;
```

Comment:
ray.hit should be initialized. As suggested in the assignment description use ray.hit = shared_from_this();